scipy/stats/tests/test_multivariate.py

```diff
@@ -35,6 +36,105 @@
 from unittest.mock import patch
 
 
+class TestCovariance:
+    def test_iv(self):
+        message = "The input `precision` must be a square, two-dimensional..."
+        with pytest.raises(ValueError, match=message):
+            stats.CovViaPrecision(np.ones(2))
+
+        message = "`precision.shape` must equal `covariance.shape`."
+        with pytest.raises(ValueError, match=message):
+            stats.CovViaPrecision(np.eye(3), covariance=np.eye(2))
+
+    def test_mean(self):
+        # test the interaction between a Covariance object and mean
+        P = np.diag(1 / np.array([1, 2, 3]))
+        cov_object = stats.CovViaPrecision(P)
+
+        message = "`cov` represents a covariance matrix in 3 dimensions..."
+        with pytest.raises(ValueError, match=message):
+            multivariate_normal.entropy([0, 0], cov_object)
+
+        x = [0.5, 0.5, 0.5]
+        ref = multivariate_normal.pdf(x, [0, 0, 0], cov_object)
+        assert_equal(multivariate_normal.pdf(x, cov=cov_object), ref)
```

Comment:
Maybe, but like above, I think it can be done with less parameterization. It is basically just a check of whether the covariance object has a valid `A` attribute.